src/api/offers.js

```diff
@@ -1,128 +1,130 @@
     validateMiddleware,
-    async (request, response) => {
+    async (request: $Request, response: $Response): Promise<void> => {
       try {
         const {token} = matchedData(request)
         const offer = await offerService.getOfferByToken(token)
 
         if (!offer) {
-          return response.status(HttpStatus.NOT_FOUND).json('No such offer')
+          response.status(HttpStatus.NOT_FOUND).json('No such offer')
         } else if (offer.isExpired() || offer.deleted) {
-          return response.status(HttpStatus.GONE).json('Offer not available')
+          response.status(HttpStatus.GONE).json('Offer not available')
         }
 
         await offerService.confirmOffer(offer, token)
-        return response.status(HttpStatus.OK).end()
+        response.status(HttpStatus.OK).end()
       } catch (e) {
         console.error(e)
-        return response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
+        response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
       }
     }
-  ])
+  )
 
-  router.post(`/:token/extend`, [
+  router.post(`/:token/extend`,
     param('token').isHexadecimal().isLength(TOKEN_LENGTH),
-    body('duration').isInt().toInt().custom(value => [3, 7, 14, 30].includes(value)),
+    body('duration').isInt().toInt().custom((value: number): boolean => [3, 7, 14, 30].includes(value)),
     validateMiddleware,
-    async (request, response) => {
+    async (request: $Request, response: $Response): Promise<void> => {
       try {
         const {token, duration} = matchedData(request)
         const offer = await offerService.getOfferByToken(token)
 
         if (!offer) {
-          return response.status(HttpStatus.NOT_FOUND).json('No such offer')
+          response.status(HttpStatus.NOT_FOUND).json('No such offer')
         } else if (offer.deleted || !offer.confirmed) {
-          return response.status(HttpStatus.BAD_REQUEST).json('Offer not available')
+          response.status(HttpStatus.BAD_REQUEST).json('Offer not available')
         }
 
         await offerService.extendOffer(offer, duration, token)
-        return response.status(HttpStatus.OK).end()
+        response.status(HttpStatus.OK).end()
       } catch (e) {
         console.error(e)
-        return response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
+        response.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e)
       }
     }
-  ])
+  )
 
-  router.delete(`/:token`, [
+  router.delete(`/:token`,
     param('token').isHexadecimal().isLength(TOKEN_LENGTH),
     validateMiddleware,
-    async (request, response) => {
+    async (request: $Request, response: $Response): Promise<void> => {
       try {
         const {token} = matchedData(request)
         const offer = await offerService.getOfferByToken(token)
 
         if (!offer) {
-          return response.status(HttpStatus.NOT_FOUND).json('No such offer')
+          response.status(HttpStatus.NOT_FOUND).json('No such offer')
```

Comment:
If you don't return here, we still reach line 120 (`await offerService.deletOffer(offer)`), right?

Reply:
I'll have a look at it :)

Reply:
Fixed it.